### PR TITLE
Fixed Android build for older NDKs

### PR DESCRIPTION
**Pull request by TGRCdev**
This PR fixes building for Android with NDK versions older than revision 19. Previously, the SConstruct file assumed that the NDK provided by the user was capable of in-place toolchain usage. Now, the SConstruct file will check for the `source.properties` file in the NDK folder, retrieve the revision number, and apply special cases depending on the NDK revision number.

This PR adds the following Scons options:
- `ndk_toolchain`: Ignores most Android options and uses the given standalone toolchain to compile.

There are a number of special cases depending on NDK revision number:
- When `ndk_version < 19`, SConstruct will construct a toolchain using `make_standalone_toolchain.py` under the directory `tempchain`. Constructed toolchains are separated and cached by NDK version, architecture, and API level (i.e. `tempchain/ndk16-armv7-18`).
- ~~When `ndk_version < 12`, SConstruct will use `make-standalone-toolchain.sh` instead (`make_standalone_toolchain.py` added in revision 12). Errors out on Windows.~~ Minimum `ndk_version` is now `17`
- When `ndk_toolchain` is defined with a pre-made standalone toolchain, SConstruct will skip toolchain construction and use `ndk_toolchain`.
- ~~If the toolchain does not contain clang scripts (gcc is used in older NDK revisions), SConstruct will attempt to use gcc instead.~~ Minimum `ndk_version` is now `17`

NOTE: ~~Constructed toolchains are not cleaned up by `scons -c`. I can add this functionality if requested.~~ Added.

This PR has been tested on Windows 10 64-bit, Lubuntu, and macOS High Sierra, with NDK revisions 10, 11, 16, and 20.

### CODE REVIEW

**Comment by piiertho** [comment]
Hi @TGRCdev ! Great job !
I only have one question.
Will toolchain be cleaned up upon reboot, like it would be in tmp ?

**Comment by TGRCdev** [comment]
No, toolchains are stored under `tempchain`, and must be deleted manually.

**Comment by piiertho** [comment]
So I think your idea to add toolchain cleanup with `scons -c` is good. It would prevent from having toolchain accumulation that takes hard drive space.

**Comment by TGRCdev** [comment]
Done. `scons -c` will now clean all toolchains under `tempchain`.

**Comment by BastiaanOlij** [comment]
Sorry for the very late reaction on this.
@TGRCdev  would you mind resolving the conflict
@bruvzg  could use a review if you'd be so kind

File: SConstruct [hunk 1]
```diff
@@ -334,13 +413,29 @@ elif env['platform'] == 'android':
...
+            # I'd use NDK version for this, but that isn't defined when the toolchain is manually set.
+        else:
+            cc = "gcc"
+            cxx = "g++"
+        env['CC'] = '"' + toolchain + "/bin/" + arch_info["tool_path"] + '-' + cc + cmd + '"'
+        env['CXX'] = '"' + toolchain + "/bin/" + arch_info["tool_path"] + '-' + cxx + cmd + '"'
+    else:
+        env['CC'] = '"' + toolchain + "/bin/clang" + '"'
+        env['CXX'] = '"' + toolchain + "/bin/clang++" + '"'
+    env['AR'] = '"' + toolchain + "/bin/" + arch_info['tool_path'] + "-ar" + '"'
```
Comment (bruvzg) [inline SConstruct]: Incremental builds won't work without:

```
    env["RANLIB"] = '"' + toolchain + "/bin/" + arch_info['tool_path'] + "-ranlib" + '"'
```

Comment (TGRCdev) [reply SConstruct]: Sorry for not responding. Added RANLIB and bumped minimum NDK version to 17.

**Comment by akien-mga** [comment]
Sorry for the delay! This would need to be redone to match the current Godot buildsystem requirements. I'm not sure we still support NDK r17, and the `3.x` branch uses C++14, the `master` branch uses C++17, so this shouldn't enforce C++11.

**Comment by TGRCdev** [comment]
It's been a long while since I worked with Godot's build system and the NDK, and since the build system has undergone massive changes, I don't think I have the development mindset to re-introduce this change (nor am I even sure if it's a wanted feature).

Unless anyone actually owns a phone with an older Android than KitKat, I'll be closing this PR.

**Pull request closed without merging**